sui/src/rpc_gateway.rs

```diff
@@ -0,0 +1,407 @@
+        gas_budget: u64,
+        object_arguments: Vec<ObjectID>,
+        shared_object_arguments: Vec<ObjectID>,
+    ) -> RpcResult<TransactionBytes> {
+        let data = async {
+            let mut gateway = self.gateway.lock().await;
+            let package_object_ref = gateway
+                .get_object_info(package_object_id)
+                .await?
+                .reference()?;
+            // Fetch the object info for the gas obj
+            let gas_obj_ref = gateway.get_object_info(gas_object_id).await?.reference()?;
+
+            // Fetch the objects for the object args
+            let mut object_args_refs = Vec::new();
+            for obj_id in object_arguments {
+                let object_ref = gateway.get_object_info(obj_id).await?.reference()?;
+                object_args_refs.push(object_ref);
+            }
+            let pure_arguments = pure_arguments
+                .iter()
+                .map(|arg| arg.to_vec())
+                .collect::<Vec<_>>();
+
+            gateway
+                .move_call(
+                    signer,
+                    package_object_ref,
+                    module,
+                    function,
+                    type_arguments,
+                    gas_obj_ref,
+                    object_args_refs,
+                    shared_object_arguments,
+                    pure_arguments,
+                    gas_budget,
+                )
+                .await
+        }
+        .await?;
+        Ok(TransactionBytes {
+            tx_bytes: data.to_bytes(),
+        })
+    }
+
+    async fn sync_account_state(&self, address: SuiAddress) -> RpcResult<()> {
+        debug!("sync_account_state : {}", address);
+        self.gateway
+            .lock()
+            .await
+            .sync_account_state(address)
+            .await?;
+        Ok(())
+    }
+
+    async fn get_total_transaction_number(&self) -> RpcResult<u64> {
+        Ok(self.gateway.lock().await.get_total_transaction_number()?)
+    }
+
+    async fn get_transactions_in_range(
+        &self,
+        start: GatewayTxSeqNumber,
+        end: GatewayTxSeqNumber,
+    ) -> RpcResult<Vec<(GatewayTxSeqNumber, TransactionDigest)>> {
+        Ok(self
+            .gateway
+            .lock()
+            .await
+            .get_transactions_in_range(start, end)?)
+    }
+
+    async fn get_recent_transactions(
+        &self,
+        count: u64,
+    ) -> RpcResult<Vec<(GatewayTxSeqNumber, TransactionDigest)>> {
+        Ok(self.gateway.lock().await.get_recent_transactions(count)?)
+    }
+}
+
+#[serde_as]
+#[derive(Serialize, Deserialize)]
+pub struct SignedTransaction {
+    #[serde_as(as = "Base64")]
+    pub tx_bytes: Vec<u8>,
+    #[serde_as(as = "Base64")]
+    pub signature: Vec<u8>,
+    #[serde_as(as = "Base64")]
+    pub pub_key: Vec<u8>,
+}
+
+impl SignedTransaction {
+    pub fn new(tx_bytes: Vec<u8>, signature: crypto::Signature) -> Self {
+        let signature_bytes = signature.as_bytes();
```

Comment:
Ditto re sig and pub key bytes, see https://github.com/MystenLabs/sui/blob/main/sui_types/src/crypto.rs#L194